tests/tests_pytorch/test_cli.py

```diff
@@ -588,6 +587,155 @@ def __init__(self, submodule1: LightningModule, submodule2: LightningModule, mai
     assert isinstance(cli.model.submodule2, BoringModel)
 
 
+class DemoModel(BoringModel):
+    def __init__(
+        self,
+        num_classes: int = 10,
+        learning_rate: float = 0.01,
+        dropout: float = 0.1,
+        backbone_hidden_dim: int = 128,
+    ):
+        super().__init__()
+        self.save_hyperparameters()
+        self.num_classes = num_classes
+        self.learning_rate = learning_rate
+        self.dropout = dropout
+        self.backbone_hidden_dim = backbone_hidden_dim
+
+
+def test_cli_args_override_checkpoint_hparams(cleandir):
+    """
+    Check priority: ckpt hparams < CLI Args
+
+    Scenario:
+    1. Save checkpoint with specific `dropout`, `backbone_hidden_dim`
+    2. Load checkpoint, but explicitly override 'learning_rate` and `backbone_hidden_dim`
+    3. Verify that `num_classes` and `dropout` is restored from ckpt,
+        but `learning_rate` and `backbone_hidden_dim` is update from the CLI arg.
+    """
+
+    # --- Phase 1: Create a base checkpoint
+    orig_hidden_dim = 256
+    orig_dropout = 0.5
+
+    save_args = [
+        "fit",
+        f"--model.dropout={orig_dropout}",
+        f"--model.backbone_hidden_dim={orig_hidden_dim}",
+        "--trainer.devices=1",
+        "--trainer.max_steps=1",
+        "--trainer.limit_train_batches=1",
+        "--trainer.limit_val_batches=1",
+        "--trainer.default_root_dir=./",
+    ]
+
+    with mock.patch("sys.argv", ["any.py"] + save_args):
+        cli = LightningCLI(DemoModel)
+
+    checkpoint_path = str(next(Path(cli.trainer.default_root_dir).rglob("*.ckpt")))
+
+    # --- Phase 2: Resume with CLI overrides ---
+    new_lr = 0.123
+    new_hidden_dim = 512
+    override_args = [
+        "predict",
+        "--trainer.devices=1",
+        f"--model.learning_rate={new_lr}",
+        f"--model.backbone_hidden_dim={new_hidden_dim}",
+        f"--ckpt_path={checkpoint_path}",
+    ]
+
+    with mock.patch("sys.argv", ["any.py"] + override_args):
+        new_cli = LightningCLI(DemoModel)
+
+    # --- Phase 3: Assertions ---
+    target_classes = 10
+    assert new_cli.model.num_classes == target_classes, (
+        f"Checkpoint restoration failed! Expected num_classes {target_classes}, got {new_cli.model.num_classes}"
+    )
```

Comment:
`num_classes=10` is the default of the model and not overridden anywhere. Maybe this assert is unnecessary?